src/TLuaInterpreter.cpp

```diff
@@ -13830,6 +13830,10 @@ void TLuaInterpreter::initLuaGlobals()
     lua_register(pGlobalLua, "getProfileTabNumber", TLuaInterpreter::getProfileTabNumber);
     lua_register(pGlobalLua, "addFileWatch", TLuaInterpreter::addFileWatch);
     lua_register(pGlobalLua, "removeFileWatch", TLuaInterpreter::removeFileWatch);
+    lua_register(pGlobalLua, "getButtonStateById", TLuaInterpreter::getButtonStateById);
+    lua_register(pGlobalLua, "getButtonStateByName", TLuaInterpreter::getButtonStateByName);
+    lua_register(pGlobalLua, "setButtonStateById", TLuaInterpreter::setButtonStateById);
+    lua_register(pGlobalLua, "setButtonStateByName", TLuaInterpreter::setButtonStateByName);
```

Comment:
Can you merge these so it's consistent with the rest of the Lua API, ie using either name or id works?

Reply:
Um, not sure that is wise - since the name of the button is currently ALSO the text that is displayed (I have plans to separate that) - what happens if the name of the button **is** a number? That is a usage that is not *totally* unreasonable from a UI point of view, *I* think?

🤔 I suppose that one could check the `lua_type(...)` of the provided argument (instead of using `lua_isnumber(...)` or `lua_isstring(...)` which will (un)happily (in this case) *coerce* from a string that is a number to that number and vice-versa). So that it has to be a number to be treated as an ID.

Note that we already have stupid implementations in some functions where an ID *number* is required but it must be provided as a **string**!

Reply:
Ah, I have found that I *can* combine the new getter functionality within the existing `getButtonState()` function - use it without arguments for the existing (only within the script for the button concerned) or provide a *number* for an ID or a *"string"* for the (last one registered with that string as it's name). The setters can be similarly combined as well.

Commits to do this will follow shortly - expected by the end of the day...